### PR TITLE
[Merged by Bors] - feat: add exception handler middleware (VF-000)

### PR DESCRIPTION
**Pull request by trs**
**Fixes or implements VF-000**

### Brief description. What is this change?

Add the new exception handler middleware for better and more accurate error responses.

Depends on: https://github.com/voiceflow/backend-utils/pull/70

### CODE REVIEW

File: lib/middlewares/project.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ class Project extends AbstractMiddleware {
       }
 
       if (!req.headers.versionID) {
-        throw new Error();
+        throw new VError('Missing version-id header', 400);
```
Comment (trs) [inline lib/middlewares/project.ts:59]: There are entire pages in datadog with `no permissions for this version at Project.attachID` 500 errors. This should quiet most of those.

Comment (jonahsnider) [reply lib/middlewares/project.ts:59]: why would anyone throw a plain Error with no message...😖

**Review by jonahsnider — APPROVED** [review]

**Pull request closed without merging**